inc/user.class.php

```diff
@@ -1050,6 +1050,25 @@ function applyRightRules() {
             unset($this->input["_ldap_rules"]);
 
             $return = true;
+         } else if (count($dynamic_profiles) == 1) {
+            $dynamic_profile = reset($dynamic_profiles);
+            $default_profile = Profile::getDefault();
```

Comment:
This should be moved inside the `if ($dynamic_profile['is_default_profile'] == true) {` block, to prevent an useless query if this condition is not reached.